ha/core/event_analyzer/parser/parser.py

```diff
@@ -187,9 +187,10 @@ def parse_event(self, msg: str) -> HealthEvent:
             resource_id = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_PAYLOAD.value][HEALTH_ATTRIBUTES.RESOURCE_ID.value]
             event_type = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_PAYLOAD.value][HEALTH_ATTRIBUTES.RESOURCE_STATUS.value]
             specific_info = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_PAYLOAD.value][HEALTH_ATTRIBUTES.SPECIFIC_INFO.value]
-            if specific_info and specific_info["generation_id"] and source == HEALTH_EVENT_SOURCES.MONITOR.value:
-                generation_id = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_PAYLOAD.value][HEALTH_ATTRIBUTES.SPECIFIC_INFO.value]["generation_id"]
-                specific_info = {"generation_id": generation_id, "pod_restart": 0}
+            if resource_type == "node":
```

Comment:
can we use a constant instead of node here ?

Reply:
Actually do We need this check ?  This check makes it explicit that generation id is to be used only for nodes so it might make sense to have it for that purpose. 
But from functionality perspective, specific info->generation_id will anyways be null in other cases right ?

Reply:
I think, its needed, because if "generation_id" is present, we need to take care of adding "pod_restart" value also. generation_id will not be present in case of disk events and in the reverse path.
Please let me know your thoughts

Reply:
thats what I am saying . generation id will not be present for disk events etc so the code will not get exercised even if line 190 is removed

Reply:
Nyways I am okay with keeping it as well

Reply:
Ok..So, keeping this. Changed to use constant for node.